Documentation/Yarn-Spec.md

```diff
@@ -0,0 +1,636 @@
+The expression must resolve to a string value and must be a string that matches the structure of node names.
+
+The behaviour of an implementing program is unspecified when asked to jump to a destination that doesn't match a title in the project.
+
+Once the jump command has been completed the current node must no longer be used, this means any dialogue, options or commands below the jump are to be ignored.
+From that point on the destination nodes contents must instead be run.
+
+`<<jump nodeName>>` is an example of a jump command, `<<jump {$chosenMurderer}>>` is an example of a jump command using an expression to determine the destination node.
+
+#### Stop
+
+The _stop_ command is for halting all progress on the story.
+Once the stop command is reached all processing on the project must halt, no additional nodes are to be loaded and run, no additional dialogue or commands are to processed.
+The stop command has only one component, the _keyword_ `stop`.
+The stop command should reset any variable or internal state back to their initial states.
+
+`<<stop>>` is the example of the stop command.
+
+#### Set
+
+The _set_ command allows variables to be given values.
+The set command has four components: the keyword, the variable, the operator and the value and must be presented in that order.
+
+The _keyword_ is the text `set`.
+The _variable_ is the name of the variable which is to have its value changed.
+The _operator_ must be the text `to`.
+The _expression_ is any expression.
+
+The following is an example of two set commands:
+```yarn
+<<set $name to "General Kenobi">>
+<<set $boldness to $boldness + 1>>
+```
+
+The set command must follow all the rules for variable naming and expressions.
+The set command must not allow setting a variable to an expression who's value is different from the type of that variable.
+
+### Flow control
+
+Flow control is a collection of commands that allow the writer to control the flow of the story.
+There are four commands which work in conjunction to support flow control:
+
+- if
+- else
+- elseif
+- endif
+
+The order of these commands is always the same and must be followed:
+
+1. if
+1. elseif
+1. else
+1. endif
+
+The if and endif must be present, the elseif and else must be optional.
+While each of these commands are their own statement they should be considered to be part of a larger flow control statement which spans multiple lines.
+
+#### if
+
+The _if_ command is the opening command of flow control and is broken up into two parts, the keyword and the expression and must be in that order.
+The _keyword_ is the text `if`.
+The _expression_ is an expression.
+The expression must resolve to a boolean.
+
+`<<if $boldness > 1>>` is an example of an if command, `<<if 1>>` is an example of an invalid if, it is invalid because the expression does not resolve to a boolean.
+
+#### elseif
+
+The _elseif_ command is an optional component of flow control and allows for additional flow to be expressed.
```

Comment:
This doesn't explain what an `elseif` command actually does (additional branching if the previous if/elseif resolved to false)